homeassistant/components/co2signal/config_flow.py

```diff
@@ -117,19 +117,26 @@ async def _validate_and_create(
         """Validate data and show form if it is invalid."""
         errors: dict[str, str] = {}
 
-        try:
-            await self.hass.async_add_executor_job(get_data, self.hass, data)
-        except InvalidAuth:
-            errors["base"] = "invalid_auth"
-        except APIRatelimitExceeded:
-            errors["base"] = "api_ratelimit"
-        except Exception:  # pylint: disable=broad-except
-            errors["base"] = "unknown"
-        else:
-            return self.async_create_entry(
-                title=get_extra_name(data) or "CO2 Signal",
-                data=data,
-            )
+        async with ElectricityMaps(token=data[CONF_API_KEY]) as em:
```

Comment:
Can you avoid creating a new aiohttp session here and use the helper ?

https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=aiohttp#platinum-
> Uses aiohttp or httpx and allows passing in websession (if making HTTP requests)

Reply:
Yea, prepared the package for this, but totally forgotten to add it here 🤦